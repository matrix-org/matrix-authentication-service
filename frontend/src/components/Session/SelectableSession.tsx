// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { Checkbox } from "@vector-im/compound-web";
import { useTranslation } from "react-i18next";

import styles from "./SelectableSession.module.css";

type Props = {
  isSelected?: boolean;
  onSelect: () => void;
};

/**
 * Wrapper to add a controlled checkbox to a session tile
 */
const SelectableSession: React.FC<React.PropsWithChildren<Props>> = ({
  isSelected,
  onSelect,
  children,
}) => {
  const { t } = useTranslation();
  return (
    <div className={styles.selectableSession}>
      <Checkbox
        className={styles.checkbox}
        kind="primary"
        onChange={onSelect}
        aria-label={t("frontend.selectable_session.label")}
        checked={isSelected}
      />
      {children}
    </div>
  );
};

export default SelectableSession;
