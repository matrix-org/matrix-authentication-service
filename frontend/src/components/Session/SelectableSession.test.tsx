// Copyright 2023 The Matrix.org Foundation C.I.C.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// @vitest-environment happy-dom

import { render, cleanup, fireEvent } from "@testing-library/react";
import { describe, it, vi, expect, afterEach } from "vitest";

import SelectableSession from "./SelectableSession";

describe("<SelectableSession />", () => {
  afterEach(cleanup);

  it("renders an unselected session", () => {
    const onSelect = vi.fn();
    const { container } = render(
      <SelectableSession isSelected={false} onSelect={onSelect}>
        <div>Test child</div>
      </SelectableSession>,
    );
    expect(container).toMatchSnapshot();
  });

  it("calls onchange when checkbox is clicked", () => {
    const onSelect = vi.fn();
    const { getByLabelText } = render(
      <SelectableSession isSelected={true} onSelect={onSelect}>
        <div>Test child</div>
      </SelectableSession>,
    );
    fireEvent.click(getByLabelText("Select session"));

    expect(onSelect).toHaveBeenCalled();
  });
});
