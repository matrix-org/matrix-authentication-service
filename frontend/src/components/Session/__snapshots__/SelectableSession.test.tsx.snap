// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SelectableSession /> > renders an unselected session 1`] = `
<div>
  <div
    class="_selectableSession_3b8f53"
  >
    <div
      class="_container_1jek6_18 _checkbox_3b8f53"
    >
      <input
        aria-label="Select session"
        class="_input_1jek6_26"
        type="checkbox"
      />
      <div
        class="_ui_1jek6_27"
      >
        <svg
          aria-hidden="true"
          class="cpd-icon"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.55003 17.575C9.4167 17.575 9.2917 17.5542 9.17503 17.5125C9.05836 17.4709 8.95003 17.4 8.85003 17.3L4.55003 13C4.3667 12.8167 4.2792 12.5792 4.28753 12.2875C4.29586 11.9959 4.3917 11.7584 4.57503 11.575C4.75836 11.3917 4.9917 11.3 5.27503 11.3C5.55836 11.3 5.7917 11.3917 5.97503 11.575L9.55003 15.15L18.025 6.67502C18.2084 6.49169 18.4459 6.40002 18.7375 6.40002C19.0292 6.40002 19.2667 6.49169 19.45 6.67502C19.6334 6.85836 19.725 7.09586 19.725 7.38752C19.725 7.67919 19.6334 7.91669 19.45 8.10002L10.25 17.3C10.15 17.4 10.0417 17.4709 9.92503 17.5125C9.80836 17.5542 9.68336 17.575 9.55003 17.575Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      Test child
    </div>
  </div>
</div>
`;
